src/util/StringUtils.h

```diff
@@ -132,7 +132,17 @@ std::pair<size_t, std::string> getUTF8Prefix(std::string_view sv,
           "Illegal UTF sequence in ad_utility::getUTF8Prefix");
     }
   }
-  return {numCodepoints, std::string(sv.data(), i)};
+  return {numCodepoints, sv.substr(0, i)};
+}
+
+inline string_view getUTF8Substring(const std::string_view str, size_t start,
```

Comment:
Get the substring from the UTF8-encoded `str` that starts at the `start`-th codepoint as has a length of `size` codepoints. If `start` >= the number of codepoints in `str`, std::out_of_range is thrown. If `start + size` >= the number of codepoints in `str` then the substring will reach until the end of `str`. This behavior is consistent with `std::string::substr`, but working on UTF-8 characters that might have multiple bytes.